### PR TITLE
KIBA-1543: Update listings to also gather from LooksRare

### PR DESCRIPTION
**Pull request by Femi-Ogunkola**

### CODE REVIEW

File: api/scripts/looksrare.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            flag = False
+            break
+        for order in responseJson['data']:
+            startDate = datetime.datetime.utcfromtimestamp(order["startTime"])
+            endDate = datetime.datetime.utcfromtimestamp(order["endTime"])
+            currentPrice = int(order["price"])
+            #Note(Femi-Ogunkola): Confirm what the signer address is
+            offererAddress = order['signer']
+            sourceId = order["hash"]
+            isValueNative = order["currencyAddress"] == "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
```
Comment (krishan711) [inline api/scripts/looksrare.py:36]: this looks strange - why are you using weth as native?

File: api/scripts/looksrare.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        'isOrderAsk':'true',
+        'collection':'0xbCe3781ae7Ca1a5e050Bd9C4c77369867eBc307e',
+        'status[]':'VALID',
+        'pagination[first]': 100,
+        'sort':'PRICE_ASC',
+    }
+    flag = True
+    listings = []
+    assetListings = []
+    while flag:
```
Comment (krishan711) [inline api/scripts/looksrare.py:29]: update this to use your new function pls

File: api/notd/token_listing_processor.py [hunk 3]
```diff
@@ -97,3 +100,48 @@ async def get_opensea_listings_for_tokens(self, registryAddress: str, tokenIds:
...
+
+    async def get_looks_rare_listings_for_collection(self, registryAddress: str) -> List[RetrievedTokenListing]:
+        listings = []
+        queryData = {
+        'isOrderAsk':'true',
+        'collection':registryAddress,
+        'status[]':'VALID',
+        'pagination[first]': 100,
+        'sort':'PRICE_DESC',
+        }
```
Comment (krishan711) [inline api/notd/token_listing_processor.py:112]:
```suggestion
        queryData = {
          'isOrderAsk':'true',
          'collection':registryAddress,
          'status[]':'VALID',
          'pagination[first]': 100,
          'sort':'PRICE_DESC',
        }
```

File: api/notd/token_listing_processor.py [hunk 4]
```diff
@@ -97,3 +100,48 @@ async def get_opensea_listings_for_tokens(self, registryAddress: str, tokenIds:
...
+        flag = True
+        assetListings = []
+        while flag:
```
Comment (krishan711) [inline api/notd/token_listing_processor.py:115]: this flag thing is not needed, just use while True

File: api/notd/token_listing_processor.py [hunk 5]
```diff
@@ -97,3 +100,48 @@ async def get_opensea_listings_for_tokens(self, registryAddress: str, tokenIds:
...
+            if len(responseJson['data']) == 0:
+                flag = False
+                break
+            for order in responseJson['data']:
+                startDate = datetime.datetime.utcfromtimestamp(order["startTime"])
+                endDate = datetime.datetime.utcfromtimestamp(order["endTime"])
+                currentPrice = int(order["price"])
+                offererAddress = order['signer']
+                sourceId = order["hash"]
+                # NOTE(Femi-Ogunkola): LooksRare seems to send eth listings with weth currency address
```
Comment (krishan711) [inline api/notd/token_listing_processor.py:127]: 👌

File: api/notd/token_listing_processor.py [hunk 6]
```diff
@@ -97,3 +100,48 @@ async def get_opensea_listings_for_tokens(self, registryAddress: str, tokenIds:
...
+                    offererAddress=offererAddress,
+                    source='LooksRare',
+                    sourceId=sourceId,
+                )
+                assetListings.append(listing)
+            queryData['pagination[cursor]'] = order['hash']
+        tokenListingDict: Dict[str, RetrievedTokenListing] = defaultdict(RetrievedTokenListing)
+        if len(assetListings) > 0:
+            for listing in assetListings:
+                tokenListingDict[listing.tokenId] = listing
```
Comment (krishan711) [inline api/notd/token_listing_processor.py:145]: you haven't implemented the ordering yet?

Comment (Femi-Ogunkola) [reply api/notd/token_listing_processor.py:145]: The results are ordered from the api call

File: api/notd/token_listing_processor.py (hunk 4 above)
Comment (Femi-Ogunkola) [reply api/notd/token_listing_processor.py:115]: So when does it know when to break the loop

File: api/notd/token_listing_processor.py (hunk 6 above)
Comment (krishan711) [reply api/notd/token_listing_processor.py:145]: ah cool

File: api/notd/token_listing_processor.py (hunk 4 above)
Comment (krishan711) [reply api/notd/token_listing_processor.py:115]: you have the break on line 120

File: api/scripts/looksrare.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import asyncio
```
Comment (krishan711) [inline api/scripts/looksrare.py:1]: you can remove this now

Comment (krishan711) [inline api/scripts/looksrare.py:1]: you can remove this now

**Review by krishan711 — APPROVED** [review]

**Pull request merged**